esrally/driver/runner.py

```diff
@@ -2495,10 +2495,13 @@ def __repr__(self, *args, **kwargs):
 class SubmitAsyncSearch(Runner):
     async def __call__(self, es, params):
         request_params = params.get("request-params", {})
+
+        # enforce wait_for_completion_timeout = 0 so there is always an id associated with the async search
+        request_params["wait_for_completion_timeout"] = 0
```

Comment:
How about checking if wait_for_completion_timeout is set in the request, then if not set it to 0, otherwise allow the current value? That would make it possible for users to override, if they would like. We should still keep a ticket open to see if we can work round it failing if the request completes synchronously

Reply:
agreed, let me add that